src/systemctl/systemctl-list-units.c

```diff
@@ -760,3 +761,196 @@ int verb_list_timers(int argc, char *argv[], void *userdata) {
 
         return r;
 }
+
+struct automount_info {
+        const char* machine;
+        const char* id;
+        char* what;
+        char* where;
+        usec_t timeout_idle_usec;
+        bool mounted;
+};
+
+static int automount_info_compare(const struct automount_info *a, const struct automount_info *b) {
+        int r;
+
+        assert(a);
+        assert(b);
+
+        r = strcasecmp_ptr(a->machine, b->machine);
+        if (r != 0)
+                return r;
+
+        return strcmp(a->where, b->where);
+}
+
+static int collect_automount_info(sd_bus* bus, const UnitInfo* info, struct automount_info *ret_info) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message* reply = NULL;
+        _cleanup_free_ char* mount = NULL, *mount_path = NULL, *where = NULL, *what = NULL, *state = NULL;
+        usec_t timeout_idle_usec;
+        BusLocator locator;
+        int r;
+
+        assert(bus);
+        assert(info);
+        assert(ret_info);
+
+        locator = (BusLocator) {
+                .destination = "org.freedesktop.systemd1",
+                .path = info->unit_path,
+                .interface = "org.freedesktop.systemd1.Automount",
+        };
+
+        r = bus_get_property_string(bus, &locator, "Where", &error, &where);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get automount target: %s", bus_error_message(&error, r));
+
+        r = bus_get_property_trivial(bus, &locator, "TimeoutIdleUSec", &error, 't', &timeout_idle_usec);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get idle timeout: %s", bus_error_message(&error, r));
+
+        r = unit_name_from_path(where, ".mount", &mount);
+        if (r < 0)
+                return r;
```

Comment:
needs to error log. 

(i.e. we want that every error is logged exactly once. This means either functions need to log about all errors themselves, or about none, and then propagate an error code that the caller can then handle/log about. Usually functions that are "main-program-like" do the former, and functions that are "library-like" do the latter, but things are blurry, and exceptions are OK. Key though is that you decide for each function what type of the two it is. This function otherwise logs about all errors, hence you should log here too, sticking to the rule)

Reply:
Done.